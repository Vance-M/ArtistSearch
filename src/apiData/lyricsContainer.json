{
  "lyrics": "I said yes to your number\r\nAnd yes to you dating me\r\nYes we can be together\r\nBut you gotta wait for me\r\nThe first time i said no\r\nIts like I never said yes\n\n\n\nI said yes we can be together\n\nYes you can stay with me\n\nBut when I say no not tonight\n\nYou actin so ungratefully\n\nThe first time I said no\n\nIts like I never said yes\n\n\n\nYou was on the wall\n\nI was with my crew\n\nYou was watching me baby\n\nI was watching you\n\nSlowly you walked over \n\nI maintained my cool\n\nYou said hello to me\n\nI said hello to you\n\n\n\nYou ask me where I'm from\n\nI asked you what you do\n\nYou some how intrigued me\n\nI thought you were so cool\n\nSomewhere between hi and goodbye\n\nI felt so comfortable\n\nI felt like we could talk all night\n\nSo i gave my number to you\n\n\n\nI said yes to your number\n\nAnd yes to you dating me\n\nYes we can be together\n\nBut you gotta wait for me\n\nThe first time I said no\n\nIts like I never said yes\n\n\n\nI said yes we can be together \n\nYes you can stay with me\n\nBut when I say no not tonight\n\nYou actin so ungratefully \n\nThe first time I said no\n\nIts like I never said yes\n\n\n\nCertainly you can call me baby\n\nI love to here from you\n\nYes of course you can come and see me boy\n\nI wanna get to know you more\n\nSure I'm feeling you\n\nNo baby, not yet\n\nWe can't take that next step\n\nWhy you getting so upset, boy you\n\nAct as though I never told you yes before\n\nYou are so ungrateful\n\nummm ummm ummm\n\n\n\nYou was at my house \n\nI was sitting on the couch\n\nYou was really buggin\n\nSo I told you to get out\n\nI had been misjudging you\n\nYou had a lotta nerve, you\n\nToo damn old to be so immature\n\nI hope you learn\n\nYou said I move to slow\n\nI showed you to the door\n\nYou said you call me later\n\nI said don't call no more\n\nIts cool if you can't wait for me\n\nI'm glad you let me know\n\nCuz you showed me your true face baby\n\nThe first time I said no\n\n\n\n[Repeat chorus until fade]"
}
