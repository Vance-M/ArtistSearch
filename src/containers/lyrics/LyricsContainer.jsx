import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import Lyrics from '../../components/lyrics/Lyrics';
import Spinner from '../../components/Spinner';
import { getLyrics } from '../../services/apiUtils';

function LyricsContainer() {
  const [lyrics, setLyrics] = useState({});
  const [loading, setLoading] = useState(true);

  const { artistName, title } = useParams();

  useEffect(() => {
    getLyrics(artistName, title)
      .then(setLyrics)
      // .catch(error => alert(error.message))
      .finally(() => setLoading(false));
  }, []);

  console.log(lyrics);

  return loading ? (
    <Spinner />
  ) : (
    <div data-testid="lyrics-p">
      <Lyrics lyrics={lyrics} />
    </div>
  );
}

export default LyricsContainer;
