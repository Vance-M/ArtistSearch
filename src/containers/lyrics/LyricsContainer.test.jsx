import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import LyricsContainer from './LyricsContainer';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import lyricsContainer from '../../apiData/lyricsContainer.json'

const server = setupServer(
  rest.get('', (req, res, ctx) => {
    return res(ctx.json(lyricsContainer));
  })
)
describe('LyricsContainer', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays a list of albums', async () => {
    render(
      <MemoryRouter>
        <LyricsContainer />
      </MemoryRouter>
    );

    screen.findByAltText('spinner');
    const pEl = await screen.findByTestId('lyrics-p')
    return waitFor(() => {
      expect(pEl).toMatchSnapshot();
    })
  });
});
