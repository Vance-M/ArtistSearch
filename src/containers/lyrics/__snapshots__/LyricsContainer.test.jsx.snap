// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LyricsContainer displays a list of albums 1`] = `Promise {}`;
